diff --git a/src/plugins/intel_cpu/thirdparty/ACLConfig.cmake b/src/plugins/intel_cpu/thirdparty/ACLConfig.cmake
index 9caba914e6..dffdf5bdd8 100644
--- a/src/plugins/intel_cpu/thirdparty/ACLConfig.cmake
+++ b/src/plugins/intel_cpu/thirdparty/ACLConfig.cmake
@@ -365,7 +365,7 @@ elseif(NOT TARGET arm_compute::arm_compute)
     endif()
 
     # Build type
-    if(CMAKE_CROSSCOMPILING)
+    if(CMAKE_CROSSCOMPILING AND NOT CPACK_GENERATOR STREQUAL "CONDA-FORGE")
         ov_arm_compute_add_option("build" "cross_compile")
     else()
         ov_arm_compute_add_option("build" "native")
