diff --git a/src/frontends/onnx/frontend/src/utils/pooling_factory.hpp b/src/frontends/onnx/frontend/src/utils/pooling_factory.hpp
index c582257b8d..d4daef7075 100644
--- a/src/frontends/onnx/frontend/src/utils/pooling_factory.hpp
+++ b/src/frontends/onnx/frontend/src/utils/pooling_factory.hpp
@@ -34,8 +34,8 @@ class PoolingFactory {
 public:
     OPENVINO_SUPPRESS_DEPRECATED_START
     explicit PoolingFactory(const Node& node);
-    OPENVINO_SUPPRESS_DEPRECATED_END
     virtual ~PoolingFactory() = default;
+    OPENVINO_SUPPRESS_DEPRECATED_END
 
     ///
     /// \brief      Creates average pooling ONNX operation.
