diff --git a/src/plugins/intel_cpu/thirdparty/ACLConfig.cmake b/src/plugins/intel_cpu/thirdparty/ACLConfig.cmake
index 2c0b7d12ee..68d63c237c 100644
--- a/src/plugins/intel_cpu/thirdparty/ACLConfig.cmake
+++ b/src/plugins/intel_cpu/thirdparty/ACLConfig.cmake
@@ -177,7 +177,7 @@ elseif(NOT TARGET arm_compute::arm_compute)
         list(APPEND ARM_COMPUTE_OPTIONS estate=64)
         if(NOT APPLE AND CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 10.2)
             # arm_sve.h header is not available on gcc older 10.2
-            list(APPEND ARM_COMPUTE_OPTIONS multi_isa=1)
+            # list(APPEND ARM_COMPUTE_OPTIONS multi_isa=1)
         endif()
     endif()
 
