diff --git a/thirdparty/dependencies.cmake b/thirdparty/dependencies.cmake
index e815baa80c..b09ef28b5a 100644
--- a/thirdparty/dependencies.cmake
+++ b/thirdparty/dependencies.cmake
@@ -73,6 +73,7 @@ if(ENABLE_INTEL_GPU)
             # - 'conan install opencl-headers'
             find_package(OpenCLHeaders QUIET)
             if(NOT OpenCLHeaders_FOUND)
+                find_package(OpenCL QUIET)
                 message(WARNING "OpenCLHeaders not found, but OpenCLICDLoader is installed. Please, install OpenCL headers")
             else()
                 set_property(TARGET OpenCL::OpenCL APPEND PROPERTY INTERFACE_LINK_LIBRARIES OpenCL::Headers)
