diff --git a/cmake/developer_package/api_validator/api_validator.cmake b/cmake/developer_package/api_validator/api_validator.cmake
index e62206bfaaa6a..0fd1f2c03a334 100644
--- a/cmake/developer_package/api_validator/api_validator.cmake
+++ b/cmake/developer_package/api_validator/api_validator.cmake
@@ -3,6 +3,20 @@
 #
 
 if(WIN32)
+    # CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION is only set when
+    # Visual Studio generators are used, but we need it
+    # when we use Ninja as well
+    if(NOT DEFINED CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION)
+        if(DEFINED ENV{WindowsSDKVersion})
+            string(REPLACE "\\" "" WINDOWS_SDK_VERSION $ENV{WindowsSDKVersion})
+            set(CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION ${WINDOWS_SDK_VERSION})
+            message(STATUS "Use ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION} Windows SDK version")
+        else()
+          message(FATAL_ERROR "WindowsSDKVersion environment variable is not set,\
+can't find Windows SDK version. Try to use vcvarsall.bat script")
+        endif()
+    endif()
+
     set(PROGRAMFILES_ENV "ProgramFiles\(X86\)")
 
     # check that PROGRAMFILES_ENV is defined, because in case of cross-compilation for Windows