diff --git a/thirdparty/CMakeLists.txt b/thirdparty/CMakeLists.txt
index a917a6b78b..19b1ee48de 100644
--- a/thirdparty/CMakeLists.txt
+++ b/thirdparty/CMakeLists.txt
@@ -423,15 +423,13 @@ if(ENABLE_OV_PADDLE_FRONTEND OR ENABLE_OV_ONNX_FRONTEND OR ENABLE_OV_TF_FRONTEND
         if(CMAKE_VERBOSE_MAKEFILE)
             set(Protobuf_DEBUG ON)
         endif()
-        # Note: we also specify 'protobuf' in NAMES because of vcpkg
-        set(find_package_protobuf_names Protobuf protobuf)
         # try to find newer version first (major is changed)
         # see https://protobuf.dev/support/version-support/ and
         # https://github.com/protocolbuffers/protobuf/commit/d61f75ff6db36b4f9c0765f131f8edc2f86310fa
-        find_package(Protobuf 4.22.0 QUIET NAMES ${find_package_protobuf_names})
+        find_package(Protobuf 4.22.0 QUIET)
         if(NOT Protobuf_FOUND)
             # otherwise, fallback to existing default
-            find_package(Protobuf 3.20.3 REQUIRED NAMES ${find_package_protobuf_names})
+            find_package(Protobuf 3.20.3 REQUIRED)
         endif()
         set(PROTOC_EXECUTABLE protobuf::protoc)
 
