diff --git a/cmake/OpenCL.cmake b/cmake/OpenCL.cmake
index df86d3434..4faf2c1c1 100644
--- a/cmake/OpenCL.cmake
+++ b/cmake/OpenCL.cmake
@@ -34,10 +34,6 @@ endif()
 set(CMAKE_REQUIRED_DEFINITIONS "-DCL_TARGET_OPENCL_VERSION=120")
 find_package(OpenCL REQUIRED)
 
-if(OpenCL_VERSION_STRING VERSION_LESS "1.2")
-    message(FATAL_ERROR
-        "OpenCL version ${OpenCL_VERSION_STRING} is not supported, must be 1.2 or greater.")
-endif()
 
 add_definitions(-DCL_TARGET_OPENCL_VERSION=120)
 
