diff --git a/cmake/templates/OpenVINOConfig.cmake.in b/cmake/templates/OpenVINOConfig.cmake.in
index 09dc382fae..a035227d9d 100644
--- a/cmake/templates/OpenVINOConfig.cmake.in
+++ b/cmake/templates/OpenVINOConfig.cmake.in
@@ -261,7 +261,7 @@ function(_ov_target_no_deprecation_error)
         else()
             set(flags "-Wno-error=deprecated-declarations")
         endif()
-        if(CMAKE_CROSSCOMPILING)
+        if(CMAKE_CROSSCOMPILING AND NOT APPLE)
             set_target_properties(${ARGV} PROPERTIES
                                   INTERFACE_LINK_OPTIONS "-Wl,--allow-shlib-undefined")
         endif()
