diff --git a/src/plugins/intel_cpu/src/nodes/mha.cpp b/src/plugins/intel_cpu/src/nodes/mha.cpp
index b6181fce0c..dfe784b9c0 100644
--- a/src/plugins/intel_cpu/src/nodes/mha.cpp
+++ b/src/plugins/intel_cpu/src/nodes/mha.cpp
@@ -1415,7 +1415,7 @@ void MHA::execute(dnnl::stream strm) {
     if (inputPrecisions[1] == Precision::FP32) {
         mhaImpl<float>();
     } else if (inputPrecisions[1] == Precision::BF16) {
-        mhaImpl<bfloat16_t>();
+        mhaImpl<dnnl::impl::bfloat16_t>();
     } else if (inputPrecisions[1] == Precision::I8) {
         mhaImpl<int8_t>();
     } else {
