diff --git a/samples/cpp/CMakeLists.txt b/samples/cpp/CMakeLists.txt
index 8849aaf47d..68b80b2336 100644
--- a/samples/cpp/CMakeLists.txt
+++ b/samples/cpp/CMakeLists.txt
@@ -100,12 +100,7 @@ endif()
 if(TARGET gflags)
     set(GFLAGS_TARGET gflags)
 elseif(gflags_required)
-    if(EXISTS /etc/debian_version)
-        set(gflags_component nothreads_static)
-    else()
-        set(gflags_component shared)
-    endif()
-    find_package(gflags QUIET OPTIONAL_COMPONENTS ${gflags_component})
+    find_package(gflags QUIET)
     if(gflags_FOUND)
         if(TARGET ${GFLAGS_TARGET})
             # nothing
