diff --git a/src/bindings/python/CMakeLists.txt b/src/bindings/python/CMakeLists.txt
index 2e41c7e780..f08bb73506 100644
--- a/src/bindings/python/CMakeLists.txt
+++ b/src/bindings/python/CMakeLists.txt
@@ -62,6 +62,7 @@ function(ov_check_python_build_conditions)
                     list(APPEND CMAKE_MODULE_PATH "${pybind11_DIR}")
                 endif()
             endif()
+            message(WARNING "Debug arm64: pybind11_FOUND: ${pybind11_FOUND}, PYTHON_VERSION_STRING: ${PYTHON_VERSION_STRING}")
             # use libraries with the same version as python itself
             set(PYBIND11_PYTHON_VERSION ${PYTHON_VERSION_STRING})
             find_host_package(PythonLibsNew ${PYBIND11_PYTHON_VERSION} EXACT ${find_package_mode})
